nodes/src/nodes/rerank_cohere/rerank_client.py

```
@@ -0,0 +1,174 @@
+        except InternalServerError as e:
+            raise RerankServerError(f'Cohere server error: {e}') from e
+
+        results = []
+        for result in response.results:
+            results.append(
+                {
+                    'index': result.index,
+                    'relevance_score': result.relevance_score,
+                    'document': documents[result.index],
+                }
+            )
```

Ruff (0.15.7)

[warning] Use a list comprehension to create a transformed list

(PERF401)